upload/upload_test.go

```diff
@@ -186,72 +184,92 @@ func (suite *TestSuite) TestFunctionality() {
 
 	configPath, err := os.CreateTemp(os.TempDir(), "s3cmd.conf")
 	if err != nil {
-		log.Panic(err)
+		suite.FailNow("failed to create s3cmd.conf", err)
 	}
 	defer os.Remove(configPath.Name()) //nolint:errcheck
 
 	err = os.WriteFile(configPath.Name(), []byte(confFile), 0600)
 	if err != nil {
-		log.Printf("failed to write temp config file, %v", err)
+		suite.FailNow("failed to write to s3cmd.conf", err)
 	}
 
 	// Create temp dir with file
 	dir, err := os.MkdirTemp(os.TempDir(), "test")
 	if err != nil {
-		log.Panic(err)
+		suite.FailNow("failed to create temp test directory", err)
 	}
 	defer os.RemoveAll(dir) //nolint:errcheck
 
 	testfile, err := os.CreateTemp(dir, "testfile")
 	if err != nil {
-		log.Panic(err)
+		suite.FailNow("failed to create test file", err)
 	}
 	err = os.WriteFile(testfile.Name(), []byte("content"), 0600)
 	if err != nil {
-		log.Printf("failed to write temp config file, %v", err)
+		suite.FailNow("failed to write to test file", err)
 	}
 	defer os.Remove(testfile.Name()) //nolint:errcheck
 
-	var str bytes.Buffer
-	log.SetOutput(&str)
+	rescuedStdout := os.Stdout
+	stdoutReader, stdoutWriter, _ := os.Pipe()
+	os.Stdout = stdoutWriter
+
+	rescuedStderr := os.Stderr
+	stderrReader, stderrWriter, _ := os.Pipe()
+	os.Stderr = stderrWriter
 
 	// Test recursive upload
 	os.Args = []string{"upload", "--force-unencrypted", "-r", dir}
 	assert.NoError(suite.T(), Upload(os.Args, configPath.Name()))
 
+	stdoutWriter.Close() //nolint:errcheck
```

Comment:
This can aslo be written like this:
```suggestion
	_ = stdoutWriter.Close()
```
To make it clear that we don't care about the closure error.